### ISSUE TITLE
Add support for Dedicated Administrator Connection

### ISSUE THREAD

**Issue by sanagama** [Microsoft/vscode-mssql#638]
- MSSQL Extension Version: 0.2.0
- VSCode Version: 1.8.1
- OS Version: Windows 10 and Ubuntu 16.04 LTS

Steps to Reproduce:
1. Enable remote admin connections on your SQL instance per: https://msdn.microsoft.com/en-us/library/ms190468.aspx
2. Run VS Code on Windows or Ubuntu
3. Create a new file and set language mode to 'SQL'
4. Press Ctrl+Shift+C to create a connection new connection profile
5. Create a DAC connection per: https://technet.microsoft.com/en-us/library/ms178068(v=sql.105).aspx

Here's what I used:
server name: admin:sanagama-ubuntu
database: (default)
Authentication type: SQL login
username: [username]
password: [password]

Result:
Status bar says ‘connecting’ for ~2 minutes then says ‘connection error’. Below is the text from the "MSSQL" output window:

[1:59:50 PM] Connecting to server "admin:sanagama-ubuntu" on document "untitled:Untitled-1".
[1:59:50 PM] Connected to server "admin:sanagama-ubuntu" on document "untitled:Untitled-1". Server information: {"serverMajorVersion":14,"serverMinorVersion":0,"serverReleaseVersion":1,"engineEditionId":3,"serverVersion":"14.0.1.246","serverLevel":"CTP1","serverEdition":"Developer Edition (64-bit)","isCloud":false,"azureVersion":0,"osVersion":"rosoft SQL Server vNext (CTP1) - 14.0.1.246 (X64) \n\tNov  1 2016 23:24:39 \n\tCopyright (c) Microsoft Corporation\n\ton Linux (Ubuntu 16.04.1 LTS)"}
[1:59:50 PM] Disconnected on document "untitled:Untitled-1"
[1:59:50 PM] Connecting to server "admin:sanagama-ubuntu" on document "untitled:Untitled-1".
[2:02:12 PM] Error connecting to server "admin:sanagama-ubuntu". Details: A connection was successfully established with the server, but then an error occurred during the pre-login handshake. (provider: TCP Provider, error: 0 - The specified network name is no longer available.)

**Comment by kevcunnane** [comment]
This is blocked on the upstream SqlClient dependency. I verified that other tools that don't have special handling of the `admin:` protocol (e.g. SSDT) work fine and that it's all handled at the driver layer. Sanjay will open a bug directly against them to track this and we can pick up in an update.

**Comment by dudanogueira** [comment]
I had a similar issue, connecting from a linux mint with vscode and mssql extension:

OS: Linux Mint 18 Sarah
VSCODE: 1.8.1 + MSSQL 0.2.0
SQL 2008 R2

[09:27:52] Connecting to server "192.0.0.4" on document "untitled:Untitled-1".
[09:31:53] Error connecting to server "192.0.0.4". Details: Connection Timeout Expired.  The timeout period elapsed during the post-login phase.  The connection could have timed out while waiting for server to complete the login process and respond; Or it could have timed out while attempting to create multiple active connections.  The duration spent while attempting to connect to this server was - [Pre-Login] initialization=999; handshake=18; [Login] initialization=0; authentication=0; [Post-Login] complete=13000; 
[09:38:22] Connecting to server "192.0.0.4" on document "untitled:Untitled-1".
[09:42:22] Error connecting to server "192.0.0.4". Details: Connection Timeout Expired.  The timeout period elapsed during the post-login phase.  The connection could have timed out while waiting for server to complete the login process and respond; Or it could have timed out while attempting to create multiple active connections.  The duration spent while attempting to connect to this server was - [Pre-Login] initialization=1001; handshake=10; [Login] initialization=0; authentication=0; [Post-Login] complete=13000; 

Is it the same error or should I open a new Issue?

Thanks!

**Comment by kevcunnane** [comment]
@dudanogueira that sounds like a separate issue as you're not using the `admin:` prefix that indicates this is a DAC connection. Searching online it appears there may be issues with .Net Core (which we use to connect & execute queries) against earlier versions of SQL Server 2008 R2 - see https://github.com/dotnet/corefx/issues/9719 and the linked issue https://github.com/dotnet/corefx/issues/6467. Based on these issues, it seems that installing SP3 solved the problem for those users.

**Comment by kevcunnane** [comment]
This is blocked on https://github.com/dotnet/corefx/issues/16396 - once this is fixed we can look at also fixing the issue in our code that means Intellisense gets the connection instead of the query executor. We likely need a special handling of `admin:` connections.

**Comment by kevcunnane** [comment]
In researching I discovered the core issue blocking this was fixed. We should take a look in our next milestone timeframe to fix the blockers on our side that hold up DAC support. I think we could make it "just work" by checking for the `admin:` prefix and never creating an Intellisense connection in that case.
